### PR TITLE
refactor output flag

### PR DESCRIPTION
**Pull request by itaysk**
related: https://github.com/aquasecurity/tracee/issues/397

this also fixes a small issue with --capture documentation 

There are a few practices here that if approved I think we should implement in the other preparers in the future.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -163,20 +147,82 @@ func main() {
...
+
+	res := tracee.OutputConfig{}
+	if len(outputSlice) == 1 && outputSlice[0] == "help" {
+		return res, fmt.Errorf(outputHelp)
+	}
+
+	for _, o := range outputSlice {
+		outputParts := strings.SplitN(o, ":", 2)
+		if outputParts[0] == "out-file" {
+			res.OutPath = outputParts[1]
```
Comment (yanivagman) [inline main.go:179]: What if the input "out-file" was given? In that case, there is no index 1 to outputParts

File: main.go [hunk 2]
```diff
@@ -163,20 +147,82 @@ func main() {
...
+		} else if outputParts[0] == "err-file" {
+			res.ErrPath = outputParts[1]
```
Comment (yanivagman) [inline main.go:181]: same

File: main.go [hunk 3]
```diff
@@ -163,20 +147,82 @@ func main() {
...
+		} else if outputParts[0] == "option" {
+			switch outputParts[1] {
```
Comment (yanivagman) [inline main.go:183]: same

File: main.go [hunk 4]
```diff
@@ -163,20 +147,82 @@ func main() {
...
+			default:
+				return res, fmt.Errorf("invalid output option: %s, use '--option help' for more info", outputParts[1])
+			}
+		} else {
+			numParts := len(outputParts)
+			if numParts == 1 {
+				res.Format = strings.TrimPrefix(outputParts[0], "format:")
+			} else if numParts == 2 {
+				res.Format = outputParts[1]
+			}
```
Comment (yanivagman) [inline main.go:201]: Is it ok to have an input of more than 2 splits? e.g.: "option:eot:3:2"

File: main.go [hunk 5]
```diff
@@ -163,20 +147,82 @@ func main() {
...
+				res.DetectSyscall = true
+			case "exec-env":
+				res.ExecEnv = true
+			default:
+				return res, fmt.Errorf("invalid output option: %s, use '--option help' for more info", outputParts[1])
+			}
+		} else {
+			numParts := len(outputParts)
+			if numParts == 1 {
+				res.Format = strings.TrimPrefix(outputParts[0], "format:")
```
Comment (yanivagman) [inline main.go:198]: if numPart is 1, the string can't start with "format:" (as it will have a size of 2 or more) - am I wrong?

File: main.go [hunk 6]
```diff
@@ -163,20 +147,82 @@ func main() {
...
 [artifact:]all                     capture all of the above artifacts.
 
-dir:/path/to/dir        path where tracee will save produced artifacts (default: /tmp/tracee).
+dir:/path/to/dir        path where tracee will save produced artifacts. the artifact will be saved into an 'out' subdirectory. (default: /tmp/tracee).
 clear-dir               clear the captured artifacts output dir before starting (default: false).
 
+Examples:
+	--capture exec                                           | capture executed files into the default output directory
+	--capture all --capture dir:/my/dir --capture clear-dir  | delete /my/dir/out and then capture all supported artifacts into it
+	--capture write=/usr/bin --capture write=/etc/*          | capture files that were written into /usr/bin directly, or anywhere below /etc
```
Comment (yanivagman) [inline main.go:224]: write=/usr/bin without "*" will not work as we don't support this syntax

File: main.go (hunk 4 above)
Comment (itaysk) [reply main.go]: I'm doing `SplitN` with 2 splits, so we can't get more then 2 items.
the example you gave will be split into: `[option eot:3:2]`

File: main.go [hunk 7]
```diff
@@ -163,20 +147,82 @@ func main() {
...
+func prepareOutput(outputSlice []string) (tracee.OutputConfig, error) {
+	outputHelp := `
+Control how and where output is printed.
+Possilbe options:
+
+[format:]{table,table-verbose,json,gob,gotemplate=/path/to/template}   output events in the specified format. for gotemplate, specify the mandatody template file
+out-file:/path/to/file                                                 write the output to a specified file. the path to the file will be created if not existing and the file will be deleted if existing (deafult: stdout)
+err-file:/path/to/file                                                 write the errors to a specified file. the path to the file will be created if not existing and the file will be deleted if existing (deafult: stderr)
+option:{eot,stacktrace,detect-syscall,exec-env}                        augment output according to given options (default: none)
+	eot                                                                add a final event that signals the end of event stream. this is an empty event with "EventName" set to the ASCII code 4, which is "End Of Transmission"
```
Comment (yanivagman) [inline main.go:159]: Alignment?

File: main.go [hunk 8]
```diff
@@ -163,20 +147,82 @@ func main() {
...
+	stack-addresses                                                    include stack memory addresses for each event
+	detect-syscall                                                     when tracing kernel functions which are not syscalls, detect and show the original syscall that called that function
+	exec-env                                                           when tracing execve/execveat, show the environment variables that were used for execution
+Examples:
+	--output json --output option:eot                        | output as json and add an EOT event
+	--output gotemplate=/path/to/my.tmpl                     | output as the provided go template
+	--output out-file:/my/out err-file:/my/err                 | output to /my/out and errors to /my/err
```
Comment (yanivagman) [inline main.go:166]: Alignment?

**Review by yanivagman — APPROVED** [review]
Only a small comment about alignment

**Pull request merged**